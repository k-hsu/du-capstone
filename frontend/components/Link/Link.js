import React from "react";
import Text from "../Text/Text";
import { useRouter } from "next/router";

const Link = ({ children, href, ...props }) => {
  const router = useRouter();

  const onLinkClick = () => {
    router.push(href);
  };
  return (
    <Text as="a" onClick={onLinkClick} cursor="pointer" {...props}>
      {children}
    </Text>
  );
};

export default Link;
