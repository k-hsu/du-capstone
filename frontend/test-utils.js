import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { renderHook } from "@testing-library/react-hooks";

function setup(jsx) {
  return {
    user: userEvent.setup(),
    ...render(jsx),
  };
}

export * from "@testing-library/react";
export { renderHook, setup };
