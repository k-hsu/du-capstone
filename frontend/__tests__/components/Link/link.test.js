import React from "react";
import { useRouter } from "next/router";
import { render, screen, userEvent } from "../../../test-utils";
import Link from "../../../components/Link/Link";

jest.mock("next/router", () => ({ useRouter: jest.fn() }));
describe("Link", () => {
  it("should render link component", () => {
    render(<Link href="/">legend of zelda</Link>);
    expect(screen.getByText("legend of zelda")).toBeInTheDocument();
  });
  it("should route to url when clicked", async () => {
    const mockRouterPush = jest.fn();
    useRouter.mockImplementation(() => ({ push: mockRouterPush }));
    render(<Link href="/link">legend of zelda</Link>);
    await userEvent.click(screen.getByText("legend of zelda"));

    expect(mockRouterPush).toBeCalledWith("/link");
  });
});
