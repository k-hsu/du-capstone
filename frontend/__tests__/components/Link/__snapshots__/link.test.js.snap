// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should render link component styles 1`] = `
<a
  className="css-1sofp4k"
  fontFamily="IBM Plex Sans"
>
  legend of zelda
</a>
`;
